codegen/scan/src/main/java/io/helidon/codegen/scan/ScanTypeInfoFactory.java

```diff
@@ -458,6 +458,18 @@ private static Set<Modifier> toModifiers(ClassMemberInfo memberInfo) {
             if (mi.isAbstract()) {
                 result.add(Modifier.ABSTRACT);
             }
+            if (mi.isSynchronized()) {
+                result.add(Modifier.SYNCHRONIZED);
+            }
+            if (mi.isNative()) {
+                result.add(Modifier.NATIVE);
+            }
+        }
+
+        if (memberInfo instanceof FieldInfo fi) {
+            if (fi.isTransient()) {
+                result.add(Modifier.TRANSIENT);
+            }
```

Comment:
isn't `if fi.isVolatile()` missing here?

Reply:
It is not available. Added a workaround using `Modifier.isVolatile(fi.getModifiers())`